app/controllers/admin/events_controller.rb

```diff
@@ -150,6 +149,34 @@ def toggle_attendance
       end
     end
 
+    ##
+    # Received a file from user
+    # Reads file and creates commercial for event
+    # File content example:
+    # EventID:MyURL
+    def mass_commercials
+      if params[:file]
+        errors = []
+
+        params[:file][:file].read.each_line do |line|
+          id = line.match(/:/).pre_match.to_i
+          url = line.match(/:/).post_match
+          event = Event.find_by(id: id)
+          if event
+            event.commercials.create!(url: url) if event.commercials.find_by(url: url).nil?
```

Comment:
I would propose to add a unique validation to commercials model. Then you can drop this line